### ISSUE TITLE
Bad data type in update_body_from_data

### ISSUE THREAD

**Issue by demmsnt** [aio-libs/aiohttp#1306]
https://github.com/KeepSafe/aiohttp/blob/master/aiohttp/client_reqrep.py#L260

i see  elif not self.chunked and isinstance(data, io.BufferedReader):

But using aiobotocore i try 

```
    resp = yield from client.put_object(Bucket=bucket,
                                        Key=key,
                                        Body=open('ANONYMIZE.zip', 'r+b'))
```

Type of Body is io.BufferedRandom 

I replace code to 

elif not self.chunked and (isinstance(data, io.BufferedReader) or isinstance(data, io.BufferedRandom)):

And that fix problem.

**Comment by asvetlov** [comment]
Got it.
Would you create a PR?
`isinstance(data, io.BufferedReader) or isinstance(data, io.BufferedRandom)` could be replaced with `isinstance(data, (io.BufferedReader, io.BufferedRandom))`.
A functional test is required (just sending a file opened with `'r+b'` mode should be enough).

**Comment by asvetlov** [comment]
Fixed by #1312
